workspaces/arborist/lib/edge.js

```diff
@@ -112,7 +112,31 @@ class Edge {
     if (node.hasShrinkwrap || node.inShrinkwrap || node.inBundle) {
       return depValid(node, this.rawSpec, this.#accept, this.#from)
     }
-    return depValid(node, this.spec, this.#accept, this.#from)
+
+    // If there's no override we just use the spec.
```

Comment:
@jdalton This is another incorrect logic that I fixed. I added lots of comments, hopefully it's clear.

Reply:
@jdalton I fixed the typo.